### DISCUSSION TITLE
The undocumented APP_PORT variable

### DISCUSSION THREAD

**Discussion by dmarcoux** [mealie-recipes/mealie#3337] [General]
Hey,

I run Mealie on my homeserver and I want to expose the application under the default HTTP port, 80. After searching through the documentation, I didn't find anything to achieve this, beside the **API_PORT** in the [Backend configuration](https://docs.mealie.io/documentation/getting-started/installation/backend-config/). So even though it says **Do not change this if you're running in Docker**, I gave it a try, setting the variable to 80. As expected, it didn't work, but then I looked at how the Dockerfile and I've found out about the **APP_PORT** variable, which I also set to 80 by passing it when the container is built and everything works as expected. I can now access Mealie through the default HTTP port.

So with this in mind, I believe **API_PORT** and **APP_PORT** should be turned into one. The documentation should document this change. The warning **Do not change this if you're running in Docker** should also be removed.

What do you think? I could send a PR to proceed with the proposed changes.

**Comment by Kuchenpirat** [comment]
If you are using docker, which is the only supported way, you could set your docker compose file to:
```
    ports:
        - "80:9000" # 
```

The number before the colon describes the local port that is used on your machine. 
This should be everything that is needed. 

But depending on you situation (if you want to have your mealie accessible by parties outside your network) i would strongly advise you use a reverse proxy including an https certificate instead of running over a unsecure http connection.

**Comment by dmarcoux** [comment]
Yes, I am running Mealie using Docker. It's only exposed in my home network, so I don't have to bother with a reverse proxy as I have a local DNS entry in my DNS resolver for the Docker container.

Still, what's the logic behind not documenting the environment variable to change the exposed port for the app?

**Reply by Kuchenpirat** [reply]
APP_PORT is for the internal container port of the api (which in our docker compose file is 9000). 

This should not be changed. I am not sure why you can now access mealie on port 80 since this port is not forwarded by the docker compose file. 

This is also the reason this is not documented. Users should not need to change it and instead change the local port from 9925 to whatsever they need as is documented with the tooltip right on the port section of the docker compose file.

**Reply by dmarcoux** [reply]
I set APP_PORT and API_PORT both to 80 and I could run Mealie on port 80. I don't use the docker compose file, it's just directly with Docker.

**Reply by Kuchenpirat** [reply]
But do you expose port 80?
So do you have something like - p 80:80 in your command?
If yes you could just - p 80:9925 and you would not need to change anything else.

**Reply by dmarcoux** [reply]
The Docker container network configuration is set to a static IP, so I cannot expose a port other than what is defined in the Dockerfile. This is how I found out about the APP_PORT here https://github.com/mealie-recipes/mealie/blob/9b4e0dd0d95d760a3bb1e1f6fc43f2da2be59302/docker/Dockerfile#L122

**Reply by Kuchenpirat** [reply]
Ok, i see why you want to do it this way now 👍

I think there are not a lot of cons against adding that to the docs with a note similar to API_PORT
